lib/matplotlib/tests/test_widgets.py

```diff
@@ -1107,17 +1107,41 @@ def test_range_slider(orientation):
 
     slider.set_val((0.2, 0.6))
     assert_allclose(slider.val, (0.2, 0.6))
+    if orientation == "vertical":
+        hpositions = [h.get_ydata()[0] for h in slider._handles]
+        assert_allclose(hpositions, (0.2, 0.6))
+    else:
+        hpositions = [h.get_xdata()[0] for h in slider._handles]
+        assert_allclose(hpositions, (0.2, 0.6))
     box = slider.poly.get_extents().transformed(ax.transAxes.inverted())
     assert_allclose(box.get_points().flatten()[idx], [0.2, .25, 0.6, .75])
 
     slider.set_val((0.2, 0.1))
     assert_allclose(slider.val, (0.1, 0.2))
+    if orientation == "vertical":
+        hpositions = [h.get_ydata()[0] for h in slider._handles]
+        assert_allclose(hpositions, (0.1, 0.2))
+    else:
+        hpositions = [h.get_xdata()[0] for h in slider._handles]
+        assert_allclose(hpositions, (0.1, 0.2))
 
     slider.set_val((-1, 10))
     assert_allclose(slider.val, (0, 1))
+    if orientation == "vertical":
+        hpositions = [h.get_ydata()[0] for h in slider._handles]
+        assert_allclose(hpositions, (0, 1))
+    else:
+        hpositions = [h.get_xdata()[0] for h in slider._handles]
+        assert_allclose(hpositions, (0, 1))
```

Comment:
Alright, I simplified the tests in the latest commit.

Reply:
I wanted to add click_and_drag testing, but I think it may be a bit difficult with the RangeSlider because of how the click event is registered. I'll keep looking into it, but I'm not sure that I'll have it ready for this PR. 

There are other widgets that look like they could also benefit from click_and_drag testing, so I may make another PR in the future addressing that.

I appreciate all of the help, Ian.

Reply:
ah indeed I see the problem with using that. I'd suggest opening a new issue about that where we can discuss how to further improve and generalize the `click_and_drag` I think it will boil down to actually sending events through the canvas like I do in mpl-playback: https://github.com/ianhi/mpl-playback/blob/2c0896205b94e2c3c1ee9e8582291cc2e94aab99/mpl_playback/playback.py#L219

Reply:
> I wanted to add click_and_drag testing, but I think it may be a bit difficult with the RangeSlider because of how the click event is registered. I'll keep looking into it, but I'm not sure that I'll have it ready for this PR.

I opened the issue #22720 to address the issues with using mouse testing with some widgets.